File: python/palindromic_substring.py
class Solution(object):
    # O(n^2) time | O(1) space
    def countSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        def countPali(i, j):
            count = 0
            while i >= 0 and j <= len(s) - 1:
                if s[i] != s[j]:
                    break
                count += 1
                i -= 1
                j += 1
            return count

        res = 0
        for i in range(len(s)):
            res += countPali(i, i)
            res += countPali(i, i + 1)
        return res
    

solution = Solution()
print(solution.countSubstrings("aaa"))

File: src/main/java/june_2025/PalindromicSubstring.java
package june_2025;

public class PalindromicSubstring {

    public static void main(String[] args) {
        String s = "aaa";

        PalindromicSubstring palindromicSubstring = new PalindromicSubstring();
        int result = palindromicSubstring.countSubstrings(s);
        System.out.println(result);

    }

    // O(n^2) time | O(1) space
    public int countSubstrings(String s) {
        int res = 0;

        for (int i = 0; i < s.length(); i++) {
            res += countPali(s, i, i);
            res += countPali(s, i, i + 1);
        }

        return res;
    }

    private int countPali(String s, int i, int j) {
        int res = 0;
        while (i >= 0 && j <= s.length() - 1) {
            if (s.charAt(i) != s.charAt(j)) {
                break;
            }
            res++;
            i--;
            j++;
        }

        return res;
    }

}
